iamy.go

```diff
@@ -85,3 +94,29 @@ func init() {
 	}
 	defaultDir = filepath.Clean(dir)
 }
+
+func performVersionChecks() {
+	currentIAMyVersion, _ := semver.Make(Version)
+	log.Printf("current version is %s", currentIAMyVersion)
+
+	if _, err := os.Stat(versionFileName); !os.IsNotExist(err) {
```

Comment:
This can be optimised a bit:

```go
fileBytes, err := ioutil.ReadFile(versionFileName)
if err != nil {
  if !os.IsNotExist(err) {
    fmt.Fprintf(os.Stderr, "could not read version file %s, ignoring: %s", versionFileName, err)
  }
  return
}
```

Reply:
I've updated this function a bit via f11beb2. I'm generally a huge advocate for returning early however in Golang, it sometimes ends up being a bit more verbose which I why I didn't initially here. It does seem more readable now so 👍